### PR TITLE
replace context menu select all/deselect all with ui element

### PR DESCRIPTION
**Pull request by mohitsuman**
#920 

- https://issues.jboss.org/browse/JBDS-4554

- This is made as per the Mockup 1 provided by crobson.  https://issues.jboss.org/secure/attachment/12425983/Selectall-Option1.png
image

### CODE REVIEW

**Comment by mohitsuman** [comment]
UI tests are passing. I have fixed the failed unit test. Will commit them.

On Oct 17, 2017 12:27 PM, "Denis Golovin" <notifications@github.com> wrote:

*@dgolovin* requested changes on this pull request.

Looks good to me, but UI tests have to be fixed.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/redhat-developer-tooling/developer-platform-install/pull/921#pullrequestreview-69780380>,
or mute the thread

File: test/unit/pages/selection/controller-test.js [hunk 1]
```diff
@@ -304,45 +304,35 @@ describe('SelectionController', function() {
...
-  });
 
-  describe('deselectAll', function() {
-    beforeEach(inject(context));
-    it('should deselect all detected components', function() {
-      selectionController.deselectAll();
+    it('should deselect all selected components', function() {
+      selectionController.isAllSelected = false;
+      selectionController.toggleSelectAll();
       for (let installer of installerDataSvc.allInstallables().values()) {
```
Comment (dgolovin) [inline test/unit/pages/selection/controller-test.js]: you should iterate over controller.checkboxModel, rater through all installables in installerDataSvc.
``` javascript
  for (let installer in selectionController.checkboxModel) {
    expect(installer.selectedOption).equals('detected');
  }
```

Comment (mohitsuman) [reply test/unit/pages/selection/controller-test.js]: Fixed this.

**Comment by mohitsuman** [comment]
button-selectall

This is the new UI layout included for Select All/ Clear All.

**Comment by mohitsuman** [comment]
test this

**Comment by mohitsuman** [comment]
fixes #920

**Pull request merged**